### ISSUE TITLE
SVG import failure with vite-plugin-svgr

### ISSUE THREAD

**Issue by haltcase** [vikejs/vike#1933]
### Description

This may be a recurrence of #353 where an extra query parameter (`&lang`) and file extension are being added to build-time imports of SVGs, breaking the ability to use `import.meta.glob` to import multiple SVGs as React components using `vite-plugin-svgr`.

### Reproduction

https://stackblitz.com/edit/vike-vite-plugin-svgr-error?file=pages%2Findex%2F%2BPage.tsx

This repro was built by:

1. Generate / install
	```shell
	npm create bati -- --react
	npm install --save-dev vite-plugin-svgr
	```

2. Create a couple small SVGs in `/assets`
3. Add `import.meta.glob` for the SVGs
4. Fail

### Expected

SVGs imported as React components per `vite-plugin-svgr` and rendered.

```jsonc
{
  "/assets/1.svg": <SVGComponent>,
  "/assets/2.svg": <SVGComponent>,
  "/assets/logo.svg": <SVGComponent>
}
```

### Actual

SVGs imported as mangled URL strings.

```jsonc
{
  "/assets/1.svg": "/assets/1.svg?react&lang.svg",
  "/assets/2.svg": "/assets/2.svg?react&lang.svg",
  "/assets/logo.svg": "/assets/logo.svg?react&lang.svg"
}
```

**Comment by brillout** [comment]
Can you publish the reproduction on GitHub?

**Comment by brillout** [comment]
It's likely an issue on Vite or vite-plugin-svgr's side. Can you reproduce the issue using Vite without Vike? Closing but let's re-open if you can make it work with Vite without Vike.

**Comment by haltcase** [comment]
Introducing Vike is what triggers this issue for me. I updated the repository so you can see the difference.

Compare
* **with Vike**: `USE_VIKE=true npm run dev`
* **without Vike**: `npm run dev`

**Comment by brillout** [comment]
The issue is that `vite-plugin-svgr` doesn't catch IDs such as `/home/rom/tmp/vike-vite-plugin-svgr-error/assets/logo.svg?react&lang.svg`: note the filename `logo.svg?react&lang.svg` while `vite-plugin-svgr` expects `logo.svg?react`.

I'm still inclined to think it's a Vite issue when using vite-plugin-svgr with SSR.

Can you create a Vite + SSR (without Vike) reproduction using https://github.com/bluwy/create-vite-extra? This will show that `vite-plugin-svgr` doesn't work with any Vite-powered SSR framework (Remix, Vike, ...). Or, if you can make it work, then I'm wrong it is actually related to Vike somehow (I'll re-open then and further dig on Vike's side). Closing but let's continue the conversation.

**Comment by haltcase** [comment]
I haven't been able to make it fail without Vike.

* **Remix**: https://github.com/haltcase/remix-svgr-issue
* **vite-extra/ssr-react-ts**: https://github.com/haltcase/vite-svgr-ssr-issue

**Comment by brillout** [comment]
Fix pre-released as `0.4.199-commit-ed4f15e`.

**Comment by haltcase** [comment]
@brillout confirming that change did make this work for me. Thanks!
